pkg/registry/core/pod/storage/eviction.go

```diff
@@ -307,7 +308,7 @@ func (r *EvictionREST) Create(ctx context.Context, name string, obj runtime.Obje
 	return &metav1.Status{Status: metav1.StatusSuccess}, nil
 }
 
-func addConditionAndDeletePod(r *EvictionREST, ctx context.Context, name string, validation rest.ValidateObjectFunc, options *metav1.DeleteOptions) error {
+func addConditionAndDeletePod(r *EvictionREST, ctx context.Context, name string, validation rest.ValidateObjectFunc, deleteOptions *metav1.DeleteOptions) error {
 	if feature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) {
 		pod, err := getPod(r, ctx, name)
```

Comment:
we should not be re-getting the pod here, we already got it in EvictionREST#Create and did logic based on it... the pod we're updating/deleting should be passed in

Reply:
something like:

1. when we're calling from a place we already have the current pod in EvictionREST#Create, pass in the pod
2. when we did not already fetch the pod, pass in nil
3. in this function, only call getPod() if the passed-in pod is nil

Reply:
from what I can tell we always have a reference to the pod, so we can just pass it